.github/workflows/main.yml

```diff
@@ -0,0 +1,48 @@
+name: "Smart Contracts Unit Tests"
+on:
+   push:
+    branches: [main]
+   pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [main]
+
+jobs:
+  unit-test:
+    name: Run unit tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Use Node 16
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Tests
+        run: npm run test
+
+  deploy:
+    name: Run deploy
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Use Node 16
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+              
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Deploy contracts
+        run: >
+          MNEMONIC=${{secrets.MNEMONIC}}
```

Comment:
As we have a default mnemonic i would remove this from the test
```suggestion
```

Reply:
done